src/shared/dissect-image.c

```diff
@@ -137,6 +137,29 @@ static int enumerator_for_parent(sd_device *d, sd_device_enumerator **ret) {
         return 0;
 }
 
+/* how many times to wait for udev to initialize the device */
+#define N_UDEV_INITIALIZTION_WAITS 200 /* approx. 10 s */
+
+static int wait_for_device_initialization(sd_device *d) {
+        for (unsigned i = 0; ; i ++) {
+                if (sd_device_get_is_initialized(d))
+                        break;
+
+                if (i > N_UDEV_INITIALIZTION_WAITS) {
+                        dev_t num;
+
+                        assert_se(sd_device_get_devnum(d, &num) >= 0);
+                        return log_info_errno(SYNTHETIC_ERRNO(-EBUSY),
+                                              "Giving up waiting for device %d:%d to be initalized...",
+                                              major(num), minor(num));
+                }
+
+                (void) usleep(50 * USEC_PER_MSEC);
```

Comment:
urks, ugly... and unnecessary, no?

we could probably abstract the wait_for_initialized() logic used in rfkill.c in some common code, and reuse it here? should be a simple excercise of moving that function to src/shared/udev-util or so, and making the subsystem/device name configurable as parameter...